android/expoview/src/main/java/host/exp/exponent/experience/ReactNativeActivity.java

```diff
@@ -159,133 +167,131 @@ public View getRootView() {
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(null);
 
-    mLayout = new FrameLayout(this);
-    setContentView(mLayout);
+    mContainerView = new FrameLayout(this);
+    // TODO (@bbarthec): unify color name R.color.splashBackground
+    mContainerView.setBackgroundColor(ContextCompat.getColor(this, R.color.splashscreen_background));
+    setContentView(mContainerView);
 
-    mContainer = new FrameLayout(this);
-    mLayout.addView(mContainer);
-    mLoadingView = new LoadingView(this);
-    if (!Constants.isStandaloneApp() || Constants.SHOW_LOADING_VIEW_IN_SHELL_APP) {
-      mContainer.setBackgroundColor(ContextCompat.getColor(this, R.color.splashBackground));
-      mLayout.addView(mLoadingView);
+    mReactContainerView = new FrameLayout(this);
+    mContainerView.addView(mReactContainerView);
+
+    // TODO (@bbarthec) if (!Constants.isStandaloneApp()) {
+    if (this.shouldCreateLadingView()) {
+      mLoadingView = new LoadingView(this);
+      mLoadingView.show();
+      mContainerView.addView(mLoadingView);
     }
 
     mDoubleTapReloadRecognizer = new DoubleTapReloadRecognizer();
     Exponent.initialize(this, getApplication());
     NativeModuleDepsProvider.getInstance().inject(ReactNativeActivity.class, this);
-    mSplashScreenKernelService = mExpoKernelServiceRegistry.getSplashScreenKernelService();
```

Comment:
I think there will be a conflict here, you may want to rebase onto master ASAP so you are testing up-to-date code.